lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -550,6 +549,10 @@ JavaScriptCompiler.prototype = {
     }
 
     this.push(this.objectLiteral(hash.values));
+    if (splat) {
+      let splatIdentifier = 'depth' + splat.depth + '.' + splat.original; // or parts[0] ? what's the different
```

Comment:
This should use one of the current lookup mechanisms, likely `lookupOnContext` combined with a `popStack` to maintain consistency in variable lookup.